### PR TITLE
TC-3037: retrieve ticketDesignator from AirPricingInfo

### PR DESCRIPTION
**Pull request by kommandant-topp**

### CODE REVIEW

File: src/utils/form-fare-basis-code.js [hunk 1]
```diff
@@ -0,0 +1,21 @@
+module.exports = (airPricingInfo, coupon) => {
+  const couponTicketDesignator = coupon['air:TicketDesignator']
+    && coupon['air:TicketDesignator'].Value;
+
+  if (!airPricingInfo && couponTicketDesignator) {
```
Comment (dchertousov) [inline src/utils/form-fare-basis-code.js:5]: IMO, nested if will be much easier to read

File: src/utils/form-fare-basis-code.js [hunk 2]
```diff
@@ -0,0 +1,21 @@
...
+    return `${coupon.FareBasis}/${couponTicketDesignator}`;
+  } if (!airPricingInfo) {
+    return coupon.FareBasis;
+  }
+
+  const [fareInfoData] = Object.values(airPricingInfo['air:FareInfo'])
```
Comment (dchertousov) [inline src/utils/form-fare-basis-code.js:11]: Is designator from `airPricingInfo` more prioritized?

File: src/utils/form-fare-basis-code.js (hunk 1 above)
Comment (dchertousov) [reply src/utils/form-fare-basis-code.js:5]: Or even ternary inside if s in the code below

**Review by dchertousov — COMMENTED** [review]
Please see my comments.
Thanks

File: src/utils/form-fare-basis-code.js (hunk 2 above)
Comment (kommandant-topp) [reply src/utils/form-fare-basis-code.js:13]: correct, according to Martin's response: `I would say that you have to get AirPricingInfo and there parse FareInfo for each segment to get AirSegment/FareBasis/TicketDesignator. And AirSegments can be paired  with flight coupons` and test results

**Review by dchertousov — APPROVED** [review]

**Review by dchertousov — APPROVED** [review]

**Pull request merged**